governance/governance.go

```diff
@@ -380,10 +380,14 @@ func (w *GovernanceWorker) start() {
 					} else if ags[0].AgreementTerminatedTime != 0 {
 						glog.Infof(logString(fmt.Sprintf("Received ReplyAck for agreement id %v. However this agreement has been terminated.", replyAck.AgreementId())))
 					} else if replyAck.ReplyAgreementStillValid() {
-						if proposal, err := protocolHandler.DemarshalProposal(ags[0].Proposal); err != nil {
-							glog.Errorf(logString(fmt.Sprintf("unable to demarshal proposal for agreement %v from database", replyAck.AgreementId())))
-						} else if err := w.RecordReply(proposal, citizenscientist.PROTOCOL_NAME); err != nil {
-							glog.Errorf(logString(fmt.Sprintf("unable to record reply %v, error: %v", *replyAck, err)))
+						if ags[0].AgreementAcceptedTime != 0 {
```

Comment:
I think you should move this check up to line 380:
} else if ags[0].AgreementAcceptedTime != 0 || ags[0].AgreementTerminatedTime != 0 {